plotting/private/bg_rgba2rgb.m

```diff
@@ -10,64 +10,89 @@
 %   rgb = bg_rgba2rgb(bg, rgba, cmap, clim, alpha, amap, alim);
 %
 % When 2 input arguments are supplied:
-%   bg   = Nx3 matrix of background rgb-coded color-values
-%   rgba = Nx4 matrix of rgb + alpha values
+%   bg   = Nx3 matrix of background rgb-coded color-values, or MxNx3
+%   rgba = Nx4 matrix of rgb + alpha values, or MxNx4
 %
 % When 7 input arguments are supplied:
-%   bg   = Nx3 matrix, Nx1 vector, or 1x3 vector.
-%   rgba = Nx1 vector with 'functional values'
+%   bg   = Nx3 matrix, Nx1 vector, 1x3 vector, MxN, or MxNx3.
+%   rgba = Nx1 vector with 'functional values', or MxN.
 %   cmap = Mx3 colormap, or MATLAB-supported name of colormap
 %   clim = 1x2 vector denoting the color limits
-%   alpha = Nx1 vector with 'alpha values'
+%   alpha = Nx1 vector with 'alpha values', or MxN
 %   amap = Mx1 alphamap, or MATLAB -supported name of alphamap ('rampup/down', 'vup/down')
 %   alim = 1x2 vector denoting the opacity limits
 
-if size(bg,1) ~= size(rgba,1)
-    error('the background and color data should have the same number of rows');
-end
-
-if size(rgba,2)==1
-    % this requires the data to be converted into rgb values first, and
-    % needs a cmap + clim, and an alpha, alphamap and alim
-    if numel(varargin)~=5,
-        error('if a vector of color data is supplied, more input arguments are required');
-    end
-    cmap = varargin{1};
-    if ischar(cmap)
-        cmap = colormap(cmap);
-    elseif size(cmap,2)~=3
-        error('invalid specification of colormap, should be nx3');
-    end
-    
-    clim = varargin{2};
-    
-    dat     = rgba;
-    finvals = isfinite(rgba);
-    
-    rgba        = zeros(size(dat,1),4);
-    rgba(finvals,1:3) = dat2rgb(dat(finvals), cmap, clim);
-    
-    alpha = varargin{3};
-    amap  = varargin{4};
-    alim  = varargin{5};
+if numel(varargin)==0
+  siz1 = size(bg);
+  siz2 = size(rgba);
+  assert(isequal(siz1(1:end-1),siz2(1:end-1)), 'number of data points should be the same');
+  assert(siz1(end)==3 && siz2(end)==4,         'inconsistent input size');
+  
+  bg   = reshape(bg,   [], 3);
+  rgba = reshape(rgba, [], 4);
+  rgb  = do_conversion(bg, rgba);
+  rgb  = reshape(rgb, siz1);
+else
+  % this requires the data to be converted into rgb values irst, and
+  % needs a cmap + clim, and an alpha, alphamap and alim
+  if numel(varargin)~=5
+    error('if a vector of color data is supplied, more input arguments are required');
+  end
+  
+  % colormap handling
+  cmap = varargin{1};
+  if ischar(cmap)
+    cmap = colormap(cmap);
+  elseif size(cmap,2)~=3
+    error('invalid specification of colormap, should be nx3');
+  end
+  clim  = varargin{2}; if isempty(clim), clim(1) = min(rgba(:)); clim(2) = max(rgba(:)); end
+  alpha = varargin{3};
+  amap  = varargin{4};
+  alim  = varargin{5}; if isempty(alim), alim(1) = min(alpha(:)); alim(2) = max(alpha(:)); end
+        
+  % deal with the color data
+  dat     = rgba;
+  siz     = size(dat);
```

Comment:
I think l. 56 needs to be `siz = size(dat, 1),` otherwise the reshape step in l. 91 outputs a rgb with a singleton dim 'npos x 1 x 3' which makes it crash in ft_plot_mesh.